packages/clay-drop-down/src/Menu.tsx

```diff
@@ -70,30 +70,54 @@ const ClayDropDownMenu = React.forwardRef<HTMLDivElement, IProps>((
 		hasLeftSymbols,
 		hasRightSymbols,
 		onSetActive,
+		style,
 		...otherProps
 	}: IProps,
 	// TS + refs don't always play nicely together, which is why it is casted
 	// in so many places below.
 	// See https://github.com/microsoft/TypeScript/issues/30748#issuecomment-480197036
 	ref
 ) => {
+	const [region, setRegion] = React.useState<{
+		left: number;
+		top: number;
+	} | null>(null);
 	const subPortalRef = useRef<HTMLDivElement | null>(null);
 
 	useDropdownCloseInteractions([alignElementRef, subPortalRef], onSetActive);
 
 	useLayoutEffect(() => {
 		if (
+			active &&
 			alignElementRef.current &&
 			(ref as React.RefObject<HTMLDivElement>).current
 		) {
-			Align.align(
-				(ref as React.RefObject<HTMLElement>).current!,
-				alignElementRef.current,
-				alignmentPosition,
-				autoBestAlign
-			);
+			let newRegion: ReturnType<typeof Align.getAlignRegion>;
+
+			if (autoBestAlign) {
+				newRegion = Align.suggestAlignBestRegion(
+					(ref as React.RefObject<HTMLElement>).current!,
+					alignElementRef.current,
+					alignmentPosition
+				).region;
+			} else {
+				newRegion = Align.getAlignRegion(
+					(ref as React.RefObject<HTMLElement>).current!,
+					alignElementRef.current,
+					alignmentPosition
+				);
+			}
+
+			setRegion(newRegion);
 		}
-	});
+	}, [autoBestAlign, active, alignElementRef.current, alignmentPosition]);
+
+	const positioning = region
+		? {
+				left: region.left,
+				top: region.top,
+		  }
```

Comment:
We can just do this:

```
region ? region : {};
```

Since the region has the same parameters.

Reply:
I didn't do that since we only want the top and left params, otherwise region returns 

```js
bottom
height
left
right
top
width
```

Reply:
Oh ok, np. Thanks for the information.

Reply:
#golf

You could write this in two lines, although I wish there was a way to do it in one... (I can't think of one, although see below):

```javascript
const {left, top} = region || {};
const positioning = {left, top};
```

(Obviously, if there is no `region`, you'll get `{left: undefined, top: undefined}` rather than `{}` but I am guessing that is ok).

One-line version that uses [optional chaining](https://github.com/tc39/proposal-optional-chaining):

```
const positioning = {left: region?.left, top: region?.top};
```

We might want to turn on the transform for that soon, seeing as the proposal is now at stage 3.

Reply:
Oh yeah, optional chaining looks much better 🤩.